Realm/RLMApp.mm

```diff
@@ -0,0 +1,178 @@
+struct CocoaNetworkTransport : public realm::app::GenericNetworkTransport {
+    static id <RLMNetworkTransporting> transport;
+
+    void send_request_to_server(const app::Request request,
+                                std::function<void(const app::Response)> completion) override {
+        // Convert the app::Request to an RLMRequest
+        auto rlmRequest = [RLMRequest new];
+        rlmRequest.url = @(request.url.data());
+        rlmRequest.body = @(request.body.data());
+        NSMutableDictionary *headers = [NSMutableDictionary new];
+        for (auto header : request.headers) {
+            headers[@(header.first.data())] = @(header.second.data());
+        }
+        rlmRequest.headers = headers;
+        rlmRequest.method = HttpMethod_toRLMHTTPMethod[request.method];
+        rlmRequest.timeoutMS = request.timeout_ms;
+
+        // Send the request through to the Cocoa level transport
+        [transport sendRequestToServer:rlmRequest completion:^(RLMResponse * _Nonnull response) {
+            std::map<std::string, std::string> bridgingHeaders;
+            for (id key in response.headers) {
+                bridgingHeaders[[key cStringUsingEncoding:NSUTF8StringEncoding]] = [response.headers[key] cStringUsingEncoding:NSUTF8StringEncoding];
+            }
+
+            // Convert the RLMResponse to an app:Response and pass downstream to
+            // the object store
+            completion(app::Response {
+                .body = [response.body cStringUsingEncoding:NSUTF8StringEncoding],
+                .headers = bridgingHeaders,
+                .http_status_code = static_cast<int>(response.httpStatusCode),
+                .custom_status_code = static_cast<int>(response.customStatusCode)
+            });
+        }];
+    }
+};
+
+id <RLMNetworkTransporting> CocoaNetworkTransport::transport = [RLMNetworkTransport new];
+
+@implementation RLMApp : NSObject
+
++(NSMutableDictionary<NSString *, RLMApp*>*)apps {
+    static NSMutableDictionary<NSString *, RLMApp*>* _apps = nil;
+    if (!_apps) {
+        _apps = [NSMutableDictionary new];
+    }
+    return _apps;
+}
+
+-(instancetype) initWithAppId:(NSString *) appId configuration:(RLMAppConfiguration *)configuration {
+    if (self = [super init]) {
+        app::App::Config boundConfiguration = {
+            .app_id = [appId cStringUsingEncoding:NSUTF8StringEncoding]
+        };
+        boundConfiguration.transport_generator = []{
+            return std::unique_ptr<app::GenericNetworkTransport>(new CocoaNetworkTransport);
+        };
+        if (configuration) {
+            if (configuration.baseURL) {
+                boundConfiguration.base_url = util::Optional<std::string>([configuration.baseURL cStringUsingEncoding:NSUTF8StringEncoding]);
+            }
+            if (configuration.transport) {
+                CocoaNetworkTransport::transport = configuration.transport;
```

Comment:
This doesn't look like it needs to be a global variable? `configuration.transport` could just be captured by the `transport_generator` lambda and then stored as a normal member of `CocoaNetworkTransport`.